sycl/doc/ControllingSYCLLanguageFeatures.md

```diff
@@ -0,0 +1,195 @@
+# Controlling SYCL/DPC++ language features
+
+This documents aims to describe a mechanism which is used to control (i.e.
+enable or disable) different SYCL/DPC++ language and library features, like
+version of SYCL/DPC++ standard to use, set of extensions to use, etc.
+
+## Controlling SYCL/DPC++ standard version
+
+In order to control the language standard to compile for, the following options
+can be used:
+
+`-sycl-std=<value>`, `--sycl-std=<value>`, `--sycl-std <value>`
+
+SYCL language standard to compile for.
+
+Possible values:
+
+- `2017`, '121', `1.2.1`, `sycl-1.2.1`: corresponds to SYCL 1.2.1
+  specification, see [SYCL Registry] for more details and the specification
+  text.
+
+  Note: setting `-sycl-std` option automatically implies a particular C++
+  standard version to be set.
+
+  **TODO**: do we need to specify which exact version is implied by which
+  `-sycl-std` value? AFAIK, SYCL spec only specifies minimum required C++
+  version, which is C++11 and we set C++17 in our implementation
+
+- `dpcpp-0.7`: corresponds to DPC++ 0.7.
+
+  Basically, `-sycl-std=dpcpp-0.7` implies support for SYCL 1.2.1 specification,
+  bunch of extension enabled and `-std=c++17`
+
+  See [oneAPI Specification] for more details and the specification text.
+
+[SYCL Registry]: https://www.khronos.org/registry/SYCL/
+[oneAPI Specification]: https://spec.oneapi.com/
+
+Note: it is possible to change C++ version independently of SYCL/DPC++ standard
+version if that is needed: `-sycl-std=1.2.1 -std=c++14`, for example.
+
+`-std=<value>`, `--std=<value>`, `--std <value>`
+
+One more way to specify SYCL/DPC++ standard version is to use general clang
+option which allows to specify language standard to compile for.
+
+Supported values (besides listed in clang documentation/help):
+
+- `sycl-1.2.1`, `sycl-2017`: corresponds to `-sycl-std=1.2.1`
+- `dpcpp-0.7`: corresponds to `-sycl-std=dpcpp-0.7`
+
+Please note that if you specify `-std` flag several times, only the latest
+value takes effect. This means, that if you want to specifiy particular C++
+standard version instead of some default one implied by SYCL/DPC++ standard,
+you have to use two separate options: `-sycl-std` and `-std`.
+
+**TODO**. Please let me know if we don't want to go with this approach at all.
+Note on motivation: clang accepts OpenCL standards via `-std` option,
+as well as `cuda` and `hip`. So, values mentioned above were added for
+consistency and for providing some additional aliases.
+
+## Controlling SYCL/DPC++ language extensions
+
+Both SYCL and DPC++ has several extensions or proposals about how to expand
```

Comment:
My understanding is that DPC++ *is* a set of extensions over SYCL 1.2.1 , so defining DPC++ as the  -sycl-std version should simply be enabling a set of extensions on top of the sycl-2015 profile?

Reply:
Some DPC++ extensions could become SYCL optional features. The sycl-2020 profile could enable some optional features if the device supports them. So any DPC++ extension that has been included on the SYCL 2020 document but it is only supported on some devices will be enabled when -sycl-std=sycl2020 is set. The -sycl-std=dpcpp-0.XX profile would only add those extensions not yet on SYCL specification document. I guess is a larger discussion than this PR.